### ISSUE TITLE
The "ctx" parameter is not actually used, leading to loss of model confidence

### ISSUE THREAD

**Issue by zhangzhao-gg** [streamer45/silero-vad-go#26]
However, in the official Silero-VAD implementation, they use a context window mechanism:
    
    def __call__(self, x, sr: int):
      context_size = 64 if sr == 16000 else 32
      if not len(self._context):
          self._context = torch.zeros(batch_size, context_size)
      # Concatenate previous context with current input
      x = torch.cat([self._context, x], dim=1)  # [64 + 512] = 576
      
      ort_inputs = {'input': x.numpy(), 'state': self._state.numpy(), 'sr': np.array(sr, dtype='int64')}
      ort_outs = self.session.run(None, ort_inputs)
      
      # Save the end of current processing as context for next call
      self._context = x[..., -context_size:]`

**Comment by Break-Neck** [comment]
Right now the problem is actually even more unexpected: we [do manage](https://github.com/streamer45/silero-vad-go/blob/bf1977c1c97c53b89c3650a826380c518cb6860f/speech/infer_linux.go#L17-L22) the ctx in _linux implementation of Infer, but [not](https://github.com/streamer45/silero-vad-go/blob/bf1977c1c97c53b89c3650a826380c518cb6860f/speech/infer_darwin.go#L15C27-L15C30) in the _darwin one!

Got here after debugging diff in prod vs dev model inference, for us the difference is huge, note difference in the blue and red lines below

<img width="606" height="466" alt="Image" src="https://github.com/user-attachments/assets/09f0b085-80fa-4176-a016-16cbac933322" />

**Comment by Break-Neck** [comment]
Actually, the Linux case became broken too after Infer method became public 😅 The context prepend happens if `currSample > 0`, buf Infer doesn't increase that value, so, if one uses detector through Infer method only, no context will ever be used.
I'll try to fix both issues later.
